### PR TITLE
docs: try harder to use the same module instances

### PR DESCRIPTION
**Pull request by fingolfin**
@ThomasBreuer this might help with issue #547

### CODE REVIEW

**Review by thofma — APPROVED** [review]

**Comment by fingolfin** [comment]
This seems to indeed fix #547 for me, but it would be best if @ThomasBreuer confirmed (or described in detail how to repro it)

**Comment by fingolfin** [comment]
Hrmm, maybe not. I applied this patch (based on one sent to me by @ThomasBreuer):
```patch
diff --git a/src/Oscar.jl b/src/Oscar.jl
index 08492f2ab..ef0be2c31 100644
--- a/src/Oscar.jl
+++ b/src/Oscar.jl
@@ -147,9 +147,16 @@ end
 
 function build_doc()
   if !isdefined(Main, :BuildDoc)
+    println("call doc_init")
     doc_init()
   end
+  println("before activate: oscardir is ", oscardir)
+  println("before activate: pkgdir(Oscar) is ", pkgdir(Oscar))
+  println("before activate: pkgdir(AbstractAlgebra) is ", pkgdir(AbstractAlgebra))
   Pkg.activate(joinpath(oscardir, "docs")) do
+    println("in activate: oscardir is ", oscardir)
+    println("in activate: pkgdir(Oscar) is ", pkgdir(Oscar))
+    println("in activate: pkgdir(AbstractAlgebra) is ", pkgdir(AbstractAlgebra))
     Base.invokelatest(Main.BuildDoc.doit, Oscar, false, true)
   end
   open_doc()

```

And then I got this (notice how `pkgdir(AbstractAlgebra)` changes!!!):
```
before activate: oscardir is /Users/mhorn/Projekte/OSCAR/Oscar.jl
before activate: pkgdir(Oscar) is /Users/mhorn/Projekte/OSCAR/Oscar.jl
before activate: pkgdir(AbstractAlgebra) is /Users/mhorn/.julia/packages/AbstractAlgebra/fzCIm
in activate: oscardir is /Users/mhorn/Projekte/OSCAR/Oscar.jl
in activate: pkgdir(Oscar) is /Users/mhorn/Projekte/OSCAR/Oscar.jl
in activate: pkgdir(AbstractAlgebra) is /Users/mhorn/Projekte/OSCAR/AbstractAlgebra.jl
```

**Comment by fingolfin** [comment]
I have improved this some more, it works OK for me now -- @ThomasBreuer please confirm

File: docs/make_work.jl [hunk 1]
```diff
@@ -76,7 +80,7 @@ function doit(strict::Bool = true, local_build::Bool = false)
 
   # remove the copied documentation agains
   for pkg in other_packages
```
Comment (ThomasBreuer) [inline docs/make_work.jl]: `# remove the copied documentation again`
`for (pkg,pkgdir) in other_packages`

**Review by ThomasBreuer — CHANGES_REQUESTED** [review]
This works for me as soon as the final loop over `other_packages` is replaced as suggested in the comment.

File: docs/make_work.jl (hunk 1 above)
Comment (fingolfin) [reply docs/make_work.jl]: Ouch, indeed, I forgot to commit that (hence it worked locally... *sigh*). Should be fixed now.

Comment (ThomasBreuer) [reply docs/make_work.jl]: Yes, thanks.
Now I am just irritated why a line in `build_doc` is claimed to be not covered by tests. I thought that either the function is not called at all (then the whole function should be marked as not covered) or this line must be reached.

**Pull request merged**